### PR TITLE
Pipe Consensus: Add load point metrics on receiver side

### PR DESCRIPTION
**Pull request by DanielWang2035**

### CODE REVIEW

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/event/common/tsfile/TsFileInsertionPointCounter.java [hunk 1]
```diff
@@ -58,14 +58,18 @@ public TsFileInsertionPointCounter(final File tsFile, final PipePattern pattern)
...
     try {
       tsFileSequenceReader = new TsFileSequenceReader(tsFile.getPath(), true, true);
-
-      filteredDeviceMeasurementMap = filterDeviceMeasurementsMapByPattern();
       allDeviceTimeseriesMetadataMap = tsFileSequenceReader.getAllTimeseriesMetadata(false);
 
-      if (shouldParsePattern) {
-        countMatchedTimeseriesPoints();
-      } else {
+      if (Objects.isNull(this.pattern)) {
```
Comment (Caideyipi) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/event/common/tsfile/TsFileInsertionPointCounter.java:63]: Also put "root" here?

**Review by Caideyipi — APPROVED** [review]

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/event/common/tsfile/TsFileInsertionPointCounter.java (hunk 1 above)
Comment (DanielWang2035) [reply iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/event/common/tsfile/TsFileInsertionPointCounter.java]: nice catch, will do

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/receiver/protocol/pipeconsensus/PipeConsensusReceiver.java [hunk 2]
```diff
@@ -417,7 +420,9 @@ private TPipeConsensusTransferResp handleTransferFileSeal(final PipeConsensusTsF
           loadFileToDataRegion(
               fileAbsolutePath,
               ProgressIndexType.deserializeFrom(ByteBuffer.wrap(req.getProgressIndex())));
+      updateWritePointCountMetrics(req.getPointCount(), fileAbsolutePath);
       pipeConsensusReceiverMetrics.recordTsFileSealLoadTimer(System.nanoTime() - endPreCheckNanos);
+
       if (status.getCode() == TSStatusCode.SUCCESS_STATUS.getStatusCode()) {
```
Comment (Pengzna) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/receiver/protocol/pipeconsensus/PipeConsensusReceiver.java:426]: move `updateWritePointCountMetrics(req.getPointCount(), fileAbsolutePath)` inside this if?

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/receiver/protocol/pipeconsensus/PipeConsensusReceiver.java [hunk 3]
```diff
@@ -530,11 +535,14 @@ private TPipeConsensusTransferResp handleTransferFileSealWithMods(
...
+      final String tsFileAbsolutePath = fileAbsolutePaths.get(1);
       final TSStatus status =
           loadFileToDataRegion(
-              fileAbsolutePaths.get(1),
+              tsFileAbsolutePath,
               ProgressIndexType.deserializeFrom(ByteBuffer.wrap(req.getProgressIndex())));
+      updateWritePointCountMetrics(req.getPointCounts().get(1), tsFileAbsolutePath);
       pipeConsensusReceiverMetrics.recordTsFileSealLoadTimer(System.nanoTime() - endPreCheckNanos);
+
       if (status.getCode() == TSStatusCode.SUCCESS_STATUS.getStatusCode()) {
```
Comment (Pengzna) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/receiver/protocol/pipeconsensus/PipeConsensusReceiver.java:546]: ditto

**Review by Pengzna — APPROVED** [review]

**Review by Caideyipi — APPROVED** [review]

**Review by SteveYurongSu — APPROVED** [review]

**Pull request merged**